KomaMRICore/src/simulation/SimulatorCore.jl

```diff
@@ -237,7 +237,8 @@ function simulate(
         sim_params_raw["type_sim_parts"] = excitation_bool
         sim_params_raw["Nblocks"] = length(parts)
         sim_params_raw["sim_time_sec"] = timed_tuple.time
-        out = signal_to_raw_data(sig, seq; phantom_name=obj.name, sys=sys, sim_params=sim_params_raw)
+        out = simulation_output(sig, seq, obj.name, sys, sim_params_raw)
+        #out = signal_to_raw_data(sig, seq; phantom_name=obj.name, sys=sys, sim_params=sim_params_raw)
```

Comment:
The `if` here should not exist, the functions should dispatch on the `return_type<:KomaSimulationOutput` type.